### PR TITLE
Propagate temporary environment variables

### LINKED ISSUE #198: Propagation of preceding environment variables
### Command name

(all)

### Completion context

```shell
CF_HOME=/tmp/foundation01 cf target -o **<TAB>
DOCKER_HOST=tcp://192.0.2.1:2375 docker start **<TAB>
GH_CONFIG_DIR=/tmp/gh gh pr view **<TAB>
GIT_DIR=/tmp/git git add **<TAB>
KUBECONFIG=/tmp/kube kubectl get **<TAB>
VAULT_ADDR=http://192.0.2.1:8200 vault list **<TAB>
```

### What happens now?

All preceding variables are ignored.

### What do you expect to complete?

Call the commands with all preceding variables when exist.

### PR DESCRIPTION
**Pull request by hexium310**
This PR resolves #198 by exporting variables as function-local with `local -x`.

### CODE REVIEW

**Comment by chitoku-k** [comment]
The current implementation doesn't do any kind of expansion and this gives some inconvenient behaviour where a user faces unintended completion being shown as in:
```zsh
$ KUBECONFIG=~/.config/kube kubectl --context=**<TAB>
$ KUBECONFIG=$XDG_CONFIG_HOME/kube kubectl --context=**<TAB>
```

**Comment by hexium310** [comment]
Parameter expansion, command substitution and arithmetic expansion are performed with `(e)`, but `~` isn't replaced with a value of `$HOME`. Using `eval` can replace it, but is it ok if it is used like this?
```zsh
eval "local -x \"${(e)var}\""
```

**Comment by chitoku-k** [comment]
It definitely causes a side effect.
image

**Comment by hexium310** [comment]
In the first place, it has been wrong. If `\"\"` is there, `~` isn't replaced.

I think a method of replacement of `~` doesn't exist...

File: src/completers/git.zsh [hunk 1]
```diff
@@ -64,8 +64,18 @@ _fzf_complete_git() {
...
+        local v
+        for v in "${${(z)@}[1, $command_pos - 1][@]}"; do
+            local name=${v%=*}
+            local cmd=${v#*=}
+            if [[ ${cmd[1]} = '~' ]]; then
+                local -x "$name=$(zsh -c "echo $cmd")"
+            else
+                local -x "$name=${(e)cmd}"
+            fi
+        done
```
Comment (hexium310) [inline src/completers/git.zsh:78]: It seems to be working in most cases, but expansion by using [dynamic named directories](https://zsh.sourceforge.io/Doc/Release/Expansion.html#Dynamic-named-directories) which is defined in .zshrc isn't worked.

Comment (chitoku-k) [reply src/completers/git.zsh:78]: As unexported shell variables are not inherited to a child zsh process, the following case does not look like being handled; I did not test this though.
```zsh
KUBECONFIG=~/.config/$CLUSTER kubectl get **<TAB>
```

**Comment by chitoku-k** [comment]
Let's just drop support for `~` and go ahead with `(e)`.

**Review by chitoku-k — APPROVED** [review]
Cool!

**Pull request merged**